exe/Wrapper.hs

```diff
@@ -46,9 +61,17 @@ main = do
       BiosMode PrintCradleType ->
           print =<< findProjectCradle
 
-      _ -> launchHaskellLanguageServer args
-
-launchHaskellLanguageServer :: Arguments -> IO ()
+      _ -> launchHaskellLanguageServer args >>= \case
+        Right () -> pure ()
+        Left err -> do
+          T.hPutStrLn stderr "*** Startup ERROR"
+          T.hPutStrLn stderr (prettyError err NoShorten)
+          putWindowMessage J.MtError (prettyError err Shorten)
+          -- Wait for 45 seconds before shutdown, so we don't spam the same message
+          -- since some LSP clients attempt to re-launch the server.
+          threadDelay (45 * 1_000_000)
```

Comment:
I think this kills VSCode "Restart Haskell Language Server" feature.

Reply:
So, can we improve that? Should we maybe wait for a message from stdin or something like that?

Reply:
Instead of a `WindowShowMessage`, send a `SWindowShowMessageRequest` with a "Restart" `MessageActionItem`

Reply:
That is a great idea, thanks!

Reply:
Unfortunately, not sure if we can do that by easily by just parsing json messages... at some point it makes more sense in my opinion to use `lsp` and install a handler.

Reply:
Why not do that?

Reply:
no reason other than thinking it *might* be overkill. I am going to use `lsp` now.

Reply:
Fixed